react/features/stream-effects/audio-mixer/AudioMixerEffect.js

```diff
@@ -0,0 +1,71 @@
+// @flow
+
+import JitsiMeetJS from '../../base/lib-jitsi-meet';
+import { MEDIA_TYPE } from '../../base/media';
+import logger from '../../base/redux/logger';
+
+/**
+ * Class Implementing the effect interface expected by a JitsiLocalTrack.
+ * The AudioMixerEffect, as the name implies, mixes two JitsiLocalTracks containing a audio track. First track is
+ * provided at the moment of creation, second is provided through the effect interface.
+ */
+export class AudioMixerEffect {
+    /**
+     * JitsiLocalTrack that is going to be mixed into the track that uses this effect.
+     */
+    _mixAudio: Object;
+
+    /**
+     * lib-jitsi-meet AudioMixer.
+     */
+    _audioMixer: Object;
+
+    /**
+     * Creates AudioMixerEffect.
+     *
+     * @param {JitsiLocalTrack} mixAudio - JitsiLocalTrack which will be mixed with the original track.
+     */
+    constructor(mixAudio: Object) {
+        if (mixAudio.getType() !== MEDIA_TYPE.AUDIO) {
+            logger.warn('AudioMixerEffect only supports audio JitsiLocalTracks; effect will not work!');
```

Comment:
we should maybe throw here instead, since it would be a programming error

Reply:
Makes sense, changed.